backend/src/packages/articles/article.service.ts

```diff
@@ -105,6 +112,54 @@ class ArticleService implements IService {
     return article.toObject();
   }
 
+  public async getUserActivity(
+    userId: number,
+  ): Promise<UserActivityResponseDto[]> {
+    const currentDate = new Date();
+    const monthsToSubtractCount = 6;
+    const sixMonthAgo = subtractMonthsFromDate(
+      currentDate,
+      monthsToSubtractCount,
+    );
+    const daysInHalfYear = getDifferenceBetweenDates(currentDate, sixMonthAgo);
+    const halfYearActivity: UserActivityResponseDto[] = [];
+
+    const userActivity = await this.articleRepository.getUserActivity({
+      userId,
+      activityFrom: sixMonthAgo.toISOString(),
+      activityTo: currentDate.toISOString(),
+    });
+
+    for (let index = 0; index <= daysInHalfYear; index++) {
```

Comment:
Can we make it with `Array.from(Array(daysInHalfYear)).map` instead of indexed loop?

Reply:
Yeah, I thought about that, but I didn't think that loops are such a bad thing )

Reply:
I have a question, should I make a separate constant even for this number 1 here:
`const halfYearActivity: UserActivityResponseDto[] = Array.from({
      length: daysInHalfYear + 1,
    })`

Reply:
They are not a bad thing - but it may be more readable + understandable (as anybody else won't have need to dig into why we need to push some values separately and why we need to bring up next iteration with `continue`)
About index increment - you may call it `INDEX_INCREMENT` and actually this is a common thing so it could be placed in `libs/constants`